genesyscloud/resource_genesyscloud_telephony_providers_edges_phone.go

```diff
@@ -530,7 +530,7 @@ func getAllPhones(_ context.Context, sdkConfig *platformclientv2.Configuration)
 
 	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
 
-	for pageNum := 1; ; pageNum++ {
+	for pageNum := 1; pageNum <= 100; pageNum++ {
```

Comment:
This is not going to work because you will only get the first 100 pages.  If there are more than 100 pages  you will prevent the original error but introduce a new bug (you can have up to 10000 records or 1000 pages).

You need to call the API first and then get the pageCount from the first call.  You then make all of the succeeding calls up to the pageCount. That is the only way you can know for sure exactly what the issue.

My name is John Carnell and I disapprove of this PR.

Reply:
Since the page size is 100 and there can be up 10,000 records doesn't that mean there can't be more 100 pages? 1,000 pages with a page size of 100 would mean 100,000 records?

Reply:
We are future-proofing :).  By calling the API first, getting the page count, and then cycling on the page count we isolate ourselves from having to make a change if the upper limit ever goes up.  As a general rule.  We should be paginating based on the page count.  I am looking through the code-base and while our iterations technically work, they leave us up to un an unbounded loop if one of our assumptions is off.